subsys/usb/device/class/cdc_acm.c

```diff
@@ -288,10 +287,13 @@ static void tx_work_handler(struct k_work *work)
 
 	LOG_DBG("Got %zd bytes from ringbuffer send to ep %x", len, ep);
 
-	usb_transfer(ep, data, len, USB_TRANS_WRITE,
-		     cdc_acm_write_cb, dev_data);
-
-	ring_buf_get_finish(dev_data->tx_ringbuf, len);
+	ret = usb_transfer(ep, data, len, USB_TRANS_WRITE, cdc_acm_write_cb, dev_data);
+	if (ret < 0) {
+		(void)ring_buf_get_finish(dev_data->tx_ringbuf, 0);
+		return;
+	}
+	dev_data->tx_ready = false;
+	(void)ring_buf_get_finish(dev_data->tx_ringbuf, len);
```

Comment:
```c
	ret = usb_transfer(ep, data, len, USB_TRANS_WRITE,
			   cdc_acm_write_cb, dev_data);
	if (ret != 0) {
		(void)ring_buf_get_finish(dev_data->tx_ringbuf, 0);
		return;
	}

	dev_data->tx_ready = false;
	(void)ring_buf_get_finish(dev_data->tx_ringbuf, len);
```

Reply:
The line break is according to [the new formatting rule](https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style)  (100 chars per line).

Reply:
Also, ret < 0 is more robust to API changes than ret != 0 as giving back the number of finished bytes might be a future way to implement. If you want a security against an unexpected positive return value I would be fine with adding an
```
 __ASSERT(ret <= 0, "Unexpected return value");
```

Reply:
> The line break is according to [the new formatting rule](https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style) (100 chars per line).

Sure ("The line length is 100 columns or fewer"), but I am asking you to change it in a minimal way without unnecessary reformatting.



> Also, ret < 0 is more robust to API changes than ret != 0 as giving back the number of finished bytes might be a future way to implement. If you want a security against an unexpected positive return value I would be fine with adding an
> 
> ```
>  __ASSERT(ret <= 0, "Unexpected return value");
> ```

No, there is no future for `usb_transfer()`, and I am asking for exactly that
```c
	if (ret != 0) {
		(void)ring_buf_get_finish(dev_data->tx_ringbuf, 0);
		return;
	}
```

Reply:
I will change it

Reply:
Changed. Also reordered the commits as the fix on cdc acm depends on the usb_transfer() fix so first transfer, then the cdc acm. The second one now states correctly that it fixes the issue.